File: tests/conformance/fixtures/go/15-diamond-deps.go
package main

import sykli "github.com/yairfalse/sykli/sdk/go"

func main() {
	p := sykli.New()

	p.Task("A").Run("echo A")
	p.Task("B").Run("echo B").After("A")
	p.Task("C").Run("echo C").After("A")
	p.Task("D").Run("echo D").After("B", "C")

	p.Emit()
}

File: tests/conformance/fixtures/elixir/15-diamond-deps.exs
use Sykli

pipeline do
  task "A" do
    run "echo A"
  end
  task "B" do
    run "echo B"
    after_ ["A"]
  end
  task "C" do
    run "echo C"
    after_ ["A"]
  end
  task "D" do
    run "echo D"
    after_ ["B", "C"]
  end
end
|> Sykli.Emitter.validate!()
|> Sykli.Emitter.to_json()
|> IO.puts()
